Remotion/ObjectBinding/Web.ClientScript/Scripts/BocReferenceValueBase.ts

```diff
@@ -0,0 +1,211 @@
+    if (isIconUpdateEnabled)
+      ArgumentUtility.CheckNotNullAndTypeIsString('controlServiceUrl', controlServiceUrl);
+    if (isIconUpdateEnabled)
+      ArgumentUtility.CheckNotNullAndTypeIsObject('iconContext', iconContext);
+    ArgumentUtility.CheckTypeIsObject('commandInfo', commandInfo);
+    ArgumentUtility.CheckTypeIsFunction('onFailure', onFailure);
+
+    var newCommand = BocReferenceValueBase.CreateCommand(oldCommand, commandInfo, businessObject);
+
+    var oldIcon = oldCommand.find('img');
+    var newIcon = BocReferenceValueBase.CreateEmptyIcon(oldIcon, newCommand.prop('title'));
+    newCommand.append(newIcon);
+
+    oldCommand.replaceWith(newCommand);
+
+    if (isIconUpdateEnabled)
+    {
+      var pageRequestManager = Sys.WebForms.PageRequestManager.getInstance();
+      if (pageRequestManager.get_isInAsyncPostBack())
+        return oldCommand;
+
+      var params: Dictionary<string> = { businessObject: JSON.stringify(businessObject) }; // TODO WebServiceUtility.Execute requires Dictionary<string>, I hope this works, the tests will tell
+      for (var propertyName in iconContext) // TODO RM-7714 - Simplify copiyng JavaScript properties in TypeScript files
+        params[propertyName] = (iconContext as any)[propertyName];
+
+      WebServiceUtility.Execute<BocReferenceValueBase_IconInformation>(
+          controlServiceUrl!,
+          'GetIcon',
+          params,
+          function (result: BocReferenceValueBase_IconInformation): void
+          {
+            BocReferenceValueBase.UpdateIconFromWebService(newCommand, newIcon, result);
+          },
+          function (err: Sys.Net.WebServiceError)
+          {
+            onFailure(err);
+            BocReferenceValueBase.ResetCommand(newCommand);
+          });
+    }
+
+    return newCommand;
+  };
+
+  public static CreateCommand (oldCommand: JQuery, commandInfo: Nullable<BocReferenceValueBase_CommandInfo>, businessObject: any): JQuery
+  {
+    ArgumentUtility.CheckNotNull('oldCommand', oldCommand);
+    ArgumentUtility.CheckTypeIsObject('commandInfo', commandInfo);
+    ArgumentUtility.CheckTypeIsString('businessObject', businessObject);
+
+    var newCommand = $('<a/>');
+
+    var tempCommandInfo: Dictionary<unknown> = {
+      href: null,
+      target: null,
+      onClick: null,
+      title: null
+    }
+    if (commandInfo != null)
+    {
+      tempCommandInfo = jQuery.extend(true, {}, commandInfo);
+      if (businessObject == null)
+      {
+        tempCommandInfo.href = null;
+        tempCommandInfo.target = null;
+        tempCommandInfo.onClick = null;
+        tempCommandInfo.title = null;
+      }
+      else if (commandInfo.href != null)
+      {
+        tempCommandInfo.href = String.format(commandInfo.href, businessObject);
+      }
+    }
+
+    var oldCommandAttributes = oldCommand[0].attributes;
+
+    for (var i = 0; i < oldCommandAttributes.length; i++)
+    {
+      var value = oldCommandAttributes[i].nodeValue;
```

Comment:
Use `const` here since we need to use it below.

```suggestion
      const value = oldCommandAttributes[i].nodeValue;
```